twilight-validate/src/command.rs

```diff
@@ -259,6 +292,105 @@ pub fn command(value: &Command) -> Result<(), CommandValidationError> {
+
+            for option in &data.options {
+                chars += chars_option(option);
+            }
+        }
+        CommandOption::String(data) => {
+            chars += longest_localization_chars(&data.name, &data.name_localizations);
+            chars += longest_localization_chars(&data.description, &data.description_localizations);
+
+            for choice in &data.choices {
+                if let CommandOptionChoice::String {
+                    name,
+                    name_localizations,
+                    value,
+                } = choice
+                {
+                    chars += longest_localization_chars(name, name_localizations) + value.len();
+                }
+            }
+        }
+        CommandOption::Integer(data) => {
+            chars += longest_localization_chars(&data.name, &data.name_localizations);
+            chars += longest_localization_chars(&data.description, &data.description_localizations);
+
+            for choice in &data.choices {
+                if let CommandOptionChoice::Int {
+                    name,
+                    name_localizations,
+                    ..
+                } = choice
+                {
+                    chars += longest_localization_chars(name, name_localizations);
+                }
+            }
+        }
+        CommandOption::Boolean(data)
+        | CommandOption::User(data)
+        | CommandOption::Role(data)
+        | CommandOption::Mentionable(data)
+        | CommandOption::Attachment(data) => {
+            chars += longest_localization_chars(&data.name, &data.name_localizations);
+            chars += longest_localization_chars(&data.description, &data.description_localizations);
+        }
+        CommandOption::Channel(data) => {
+            chars += longest_localization_chars(&data.name, &data.name_localizations);
+            chars += longest_localization_chars(&data.description, &data.description_localizations);
+        }
+        CommandOption::Number(data) => {
+            chars += longest_localization_chars(&data.name, &data.name_localizations);
+            chars += longest_localization_chars(&data.description, &data.description_localizations);
+        }
+    }
+
+    chars
+}
+
+/// Calculate the characters for the longest name/description.
+///
+/// Discord only counts the longest localization to the character
+/// limit. If the default value is longer than any of the
+/// localizations, the length of the default value will be used
+/// instead.
+fn longest_localization_chars(
+    default: &str,
+    localizations: &Option<HashMap<String, String>>,
+) -> usize {
+    let mut chars = default.len();
+
+    if let Some(localizations) = localizations {
+        for localization in localizations.values() {
+            if localization.len() > chars {
+                chars = localization.len();
+            }
+        }
```

Comment:
Something like `localizations.values().map(|l| l.len()).max()` could work

Reply:
i think this is fine as it is